sdk/python/foundation-models/system/inference/text-to-image/aacs-prepare-uai.ipynb

```diff
@@ -0,0 +1,434 @@
+    "            role_definition_id=role_def.id,\n",
+    "            principal_id=uai_principal_id,\n",
+    "            principal_type=\"ServicePrincipal\",\n",
+    "        ),\n",
+    "    )\n",
+    "except ResourceExistsError as ex:\n",
+    "    pass\n",
+    "except Exception as ex:\n",
+    "    print(ex)\n",
+    "    raise ex\n",
+    "\n",
+    "if role_name in uai_role_check_list:\n",
+    "    del uai_role_check_list[role_name]\n",
+    "print(\"Role assignment for AcrPull at the workspace container registry completed.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.4.6 Assign Storage Blob Data Reader at the workspace storage account scope"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "role_name = \"Storage Blob Data Reader\"\n",
+    "blob_scope = workspace_resource.storage_account\n",
+    "\n",
+    "role_defs = role_definition_client.role_definitions.list(scope=blob_scope)\n",
+    "role_def = next((r for r in role_defs if r.role_name == role_name))\n",
+    "\n",
+    "from azure.core.exceptions import ResourceExistsError\n",
+    "\n",
+    "try:\n",
+    "    role_assignment_client.role_assignments.create(\n",
+    "        scope=blob_scope,\n",
+    "        role_assignment_name=str(uuid.uuid4()),\n",
+    "        parameters=RoleAssignmentCreateParameters(\n",
+    "            role_definition_id=role_def.id,\n",
+    "            principal_id=uai_principal_id,\n",
+    "            principal_type=\"ServicePrincipal\",\n",
+    "        ),\n",
+    "    )\n",
+    "except ResourceExistsError as ex:\n",
+    "    pass\n",
+    "except Exception as ex:\n",
+    "    print(ex)\n",
+    "    raise ex\n",
+    "\n",
+    "if role_name in uai_role_check_list:\n",
+    "    del uai_role_check_list[role_name]\n",
+    "print(\n",
+    "    \"Role assignment for `Storage Blob Data Reader` at the workspace storage account completed.\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2.4.7 Let's make sure we didn't miss anything in the previous steps, please execute the following script to check on that:__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check everything is properly done before creating the Azure AI Content Safety Enabled text-to-image online endpoint\n",
+    "missing_steps = []\n",
+    "print(\n",
+    "    \"You selected UAI to deploy the Azure AI Content Safety Enabled text-to-image online endpoint, checking if the UAI has the required roles assigned...\"\n",
+    ")\n",
+    "if uai_role_check_list:\n",
+    "    for key, value in uai_role_check_list.items():\n",
+    "        missing_steps.append(\n",
+    "            f'Please go to step {value[\"step\"]} to {value[\"description\"]}'\n",
```

Comment:
1. In the Markdown of cells above provide id as `<a id='some_id'></a>`
2. Here when printing reference to cells print below
```
from IPython.display import display, HTML
display(HTML("<a href=\"#some_id\">Step some_id</a>"))
```

This will create a clickable link for user to directly navigate to that section.

Reply:
Good idea.

This notebook is direct copy of the same notebook from LLama folder. This way of authentication is very involved and it's expected to be used very rarely. I think, let's keep it as it is (not to just overkill). I'm resolving it, but if you feel otherwise, please reopen it.

Reply:
ok, sure. It is not a blocking comment, just a suggestion for better user experience. Feel free to move ahead. 😊